scene/gui/item_list.cpp

```diff
@@ -71,6 +71,28 @@ int ItemList::add_item(const String &p_item, const Ref<Texture2D> &p_texture, bo
 	return item_id;
 }
 
+void ItemList::insert_item(int position, const String &p_item, const Ref<Texture2D> &p_texture, bool p_selectable) {
+	if (position == items.size()) {
+		add_item(p_item, p_texture, p_selectable);
+		return;
+	}
+	ERR_FAIL_INDEX(position, items.size());
```

Comment:
```suggestion
	ERR_FAIL_INDEX(position, items.size() + 1);
```
`Vector::insert` supports appending if the index is equal to the length

Reply:
Great catch!